src/EditorFeatures/CSharpTest/Completion/CompletionProviders/DeclarationNameCompletionProviderTests.cs

```diff
@@ -0,0 +1,462 @@
+        public async void WordBreaking2()
+        {
+            var markup = @"
+interface I {}
+public class C
+{
+    I $$
+}
+";
+            await VerifyItemExistsAsync(markup, "I");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void WordBreaking3()
+        {
+            var markup = @"
+interface II {}
+public class C
+{
+    I $$
+}
+";
+            await VerifyItemExistsAsync(markup, "I");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void WordBreaking4()
+        {
+            var markup = @"
+interface IFoo {}
+public class C
+{
+    IFoo $$
+}
+";
+            await VerifyItemExistsAsync(markup, "Foo");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void WordBreaking5()
+        {
+            var markup = @"
+class AWonderfullyLongClassName {}
+public class C
+{
+    AWonderfullyLongClassName $$
+}
+";
+            await VerifyItemExistsAsync(markup, "A");
+            await VerifyItemExistsAsync(markup, "AWonderfully");
+            await VerifyItemExistsAsync(markup, "AWonderfullyLong");
+            await VerifyItemExistsAsync(markup, "AWonderfullyLongClass");
+            await VerifyItemExistsAsync(markup, "Name");
+            await VerifyItemExistsAsync(markup, "ClassName");
+            await VerifyItemExistsAsync(markup, "LongClassName");
+            await VerifyItemExistsAsync(markup, "WonderfullyLongClassName");
+            await VerifyItemExistsAsync(markup, "AWonderfullyLongClassName");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void Parameter1()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    void Foo(CancellationToken $$
+}
+";
+            await VerifyItemExistsAsync(markup, "cancellationToken", glyph: (int)Glyph.Parameter);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void Parameter2()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    void Foo(int x, CancellationToken c$$
+}
+";
+            await VerifyItemExistsAsync(markup, "cancellationToken", glyph: (int)Glyph.Parameter);
+        }
+
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void Parameter3()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    void Foo(CancellationToken c$$) {}
+}
+";
+            await VerifyItemExistsAsync(markup, "cancellationToken", glyph: (int)Glyph.Parameter);
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Completion)]
+        public async void SuggestionsForInt()
+        {
+            var markup = @"
+using System.Threading;
+public class C
+{
+    int $$
+}
+";
+            await VerifyItemExistsAsync(markup, "v");
```

Comment:
Hm, wonder if people will find it annoying to not have them from a muscle memory perspective. Guess we'll find out.